apps/examples/nettest/nettest_stress.c

```diff
@@ -0,0 +1,376 @@
+				ret = send(g_sockfd, buf + send_sum, sbuf_size - send_sum, 0);
+				if (ret <= 0) {
+					if (ret == 0) {
+						printf("[SEND] connection closed\n");
+						goto terminate;
+					}
+					printf("[SEND] socket error ret(%d) err(%d)\n", ret, errno);
+					if (errno == 104 || errno == 9) {
+						retry_sender = 1;
+					}
+					goto terminate;
+				}
+				send_sum += ret;
+//				printf("DEBUG!! %d %d sbuf %d\n", ret, send_sum, sbuf_size);
+			}
+			printf("[SEND] send msg #%d nbytes=%d\n", i+1, send_sum);
+		}
+		printf("[SEND] Wait Response\n");
+		sending_now = 0;
+		sem_wait(&g_sem_signal);
+	}
+
+	printf("[SEND] Closed Successfully\n");
+
+	sending_now = 0;
+	return 0;
+
+terminate:
+	printf("[SEND] Closed due to fail\n");
+	return 0;
+}
+
+void *receiver_thread(void *arg)
+{
+	int count = 0;
+	int recv_len = 0;
+	int nbytes = 0;
+	int nbytes_sum = 0;
+	int ret;
+	char msg[BUF_SIZE];
+	fd_set rfds;
+	fd_set fds;
+
+	FD_ZERO(&fds);
+	FD_SET(g_sockfd, &fds);
+
+	recv_len = BUF_SIZE;
+	while (!terminate_flag) {
+
+		struct timeval timeout = { .tv_usec = SELECT_TIME_OUT };
+		rfds = fds;
+		ret = select(g_sockfd + 1, &rfds, NULL, NULL, &timeout);
+		printf("[RECV] Select out (timeout? %d)\n", ret);
+		if (ret < 0) {
+			printf("[RECV] Select failed (%d)\n", errno);
+			sem_post(&g_sem_signal);
+			close(g_sockfd);
+			return 0;
+		}
+
+
+		if (FD_ISSET(g_sockfd, &rfds)) {
+			printf("[RECV] - required data size %d\n", BUF_SIZE - nbytes_sum);
+			nbytes = recv(g_sockfd, msg + nbytes_sum, recv_len - nbytes_sum, 0);
+			if (nbytes <= 0) {
+				printf("[RECV] received failed (errno=%d, nbytes=%d)\n", errno, nbytes);
+				if (errno == 104) {
+					retry_receiver = 1;
+				} else if (errno == 11) {
+					continue;
+				}
+				close(g_sockfd);
+				sem_post(&g_sem_signal);
+				return 0;
+			}
+			nbytes_sum += nbytes;
+			if (nbytes_sum < BUF_SIZE) {
+				continue;
+			}
+			nbytes_sum = 0;
+
+			count++;
+			printf("[RECV] - Received Msg # %d] Received Msg (%d)\n", count, nbytes_sum);
+			if (count == PKT_NUM) {
+				sem_post(&g_sem_signal);
+				count = 0;
+			}
+		}
+	}
+	printf("[RECV] Closed successfully \n");
+	sem_post(&g_sem_signal);
+	close(g_sockfd);
+
+	return 0;
+}
+
+void *disconnector_thread(void *arg)
+{
+	printf("[DISC] SHUTDOWN START\n");
+	shutdown(g_sockfd, SHUT_RDWR);
+	printf("[DISC] SHUTDOWN DONE\n");
+#if PRINT_HEAPINFO
+	disconnect_count++;
+	if (disconnect_count == 10) {
+		heapinfo_parse(BASE_HEAP, HEAPINFO_SIMPLE, HEAPINFO_PID_ALL);
+		disconnect_count = 0;
+	}
+#endif
+	terminate_flag = 1;
+	return 0;
+}
+
+void nettest_stress(char *t_addr, int port)
+{
+	pthread_t sender;
+	pthread_t receiver;
+	pthread_t disconnector;
+	int ret = 0;
+	int iter = 0;
+	int connect_trial = 0;
+	int addrlen = sizeof(struct sockaddr_in);
+
+#if WIFI_AUTO_CONNECT
+	/* Connect Info
+	 * SSID : Gorani
+	 * PASS : jonbeo1@
+	 */
+	wifi_manager_result_e wret = WIFI_MANAGER_FAIL;
+	wret = wifi_manager_init(&wifi_callbacks);
```

Comment:
Is it okay not to deinit when this test is finished?

Reply:
Modified. Thanks for the comment.